simnet_tests/run_tests.sh

```diff
@@ -0,0 +1,126 @@
+# e.g: ./${SCRIPT_NAME} test-name parity/substrate:latest logs quick
+
+set -eou pipefail
+SCRIPT_NAME="$0"
+SCRIPT_PATH=$(dirname "${SCRIPT_NAME}")     # relative
+SCRIPT_PATH=$(cd "${SCRIPT_PATH}" && pwd)   # absolutized and normalized
+
+function random_string {
+   head -1 <(fold -w 30  <(tr -dc 'a-z0-9' < /dev/urandom))
+ }
+
+#
+### Script args
+#
+
+NAMESPACE=${1:-gurke-"$(random_string)"-runtest}
+IMAGE=${2:-"parity/substrate:latest"}
+LOG_PATH=${3:-"${SCRIPT_PATH}/logs"}
+FEATURES=${4:-"ALL"}
+
+mkdir -p "${SCRIPT_PATH}"/logs
+
+echo "Running tests in namespace: ${NAMESPACE}"
+echo "Testing image: ${IMAGE}"
+echo "Storing scripts logs to: ${LOG_PATH}"
+echo "Using features files from: ${FEATURES}"
+
+#
+### Script logic
+#
+
+function forward_port {
+  # RUN_IN_CONTAINER is env var that is set in the dockerfile
+  # use the -v operator to explicitly test if a variable is set
+  if  [[ ! -v RUN_IN_CONTAINER  ]]  ; then 
+    if is_port_forward_running ; then
+      kill_previous_job
+    fi
+  fi
+  start_forwading_job 
+}
+
+FORWARD_GREP_FILTER='kubectl.*[p]ort-forward.*svc/rpc.*11222'
+
+function is_port_forward_running {
+  # shellcheck disable=SC2009
+  ps aux | grep -qE "${FORWARD_GREP_FILTER}" 
+}
+
+function kill_previous_job {
+  # shellcheck disable=SC2009
+  job_pid=$(ps aux | grep -E "${FORWARD_GREP_FILTER}" | awk '{ print $2 }')
+  echo  "INFO Killed forwading port 9944 into bootnode"
+  kill "${job_pid}"
+}
+
+function start_forwading_job {
+  kubectl -n "${NAMESPACE}" \
+          expose pod bootnode \
+          --name=rpc \
+          --type=NodePort \
+          --target-port=9944 \
+          --port=9944
+  kubectl -n "${NAMESPACE}" \
+          port-forward svc/rpc 11222:9944 &> "${LOG_PATH}/forward-${NAMESPACE}.log" &
+  sleep 2
+  echo "INFO Started forwading port 9944 into bootnode"
+}
+
+function update_api {
+  echo "INFO: Updating Polkadot JS API"
+  pwd
+  cd "${SCRIPT_PATH}"/../../sub-flood/
+  npm run build
+  cd -
+}
+
+function run_test {
+    case "${FEATURES}" in
+      quick)         
+        gurke test "${NAMESPACE}" "${SCRIPT_PATH}"/tests/quick  --log-path "${LOG_PATH}"
+        ;;
+      long)  
+        gurke test "${NAMESPACE}" "${SCRIPT_PATH}"/tests/long   --log-path "${LOG_PATH}"
+        ;;
+      ALL )  
+        gurke test "${NAMESPACE}" "${SCRIPT_PATH}"/tests        --log-path "${LOG_PATH}"
+        ;;  
+      ??* )  
+        gurke test \
+              "${NAMESPACE}" \
+              "${SCRIPT_PATH}"/"${FEATURES}" \
+              --log-path  "${LOG_PATH}"
+        ;;  
+    esac
+}
+
+
+export NAMESPACE="${NAMESPACE}"
+
+set -x  # echo the commands to stdout
+gurke spawn --config "${SCRIPT_PATH}"/configs/default_local_testnet.toml \
+            -n "${NAMESPACE}" \
+            --image "${IMAGE}"
+
+echo "INFO: Checking if pods launched correctly"
+kubectl -n "${NAMESPACE}" get pods -o wide
+
+update_api
+
+forward_port
+run_test
+
+
```

Comment:
Is this script a minimum that's required to launch simnet? In order to lower the maintenance, is it possible to move its' parts to simnet repo?

Reply:
No, because this script sometimes is used for the setup of the tests, when we have no config options in the config files

Reply:
it combines three functions then: configure, launch and add some testing features. While it should only pass the payload to simnet. Launching mechanism should be on simnet side, testing should happen in test files.

Reply:
@TriplEight can you please elaborate? Right now this script is the entry point for the set of tests. Generally it has the following sections:

- Collect required variables
- Spawn the chain, using config files
- Perform additional setup for the chain (things, that cannot be done inside configuration)
- Run set of tests

So this script binds configuration and testing logic into one "entity", and CI runs this "entity". What parts do you suggest to move?

Reply:
As a tool, it simnet should have an entrypoint, the user should pass the requirements to the entrypoint, not triggering it in three steps. I suggest on the side of CI we store only the configs and tests, collect them and send to simnet.
And simnet will
- accept tests and configs
- spawn the chain (setup)
- initiate the tests
- teardown

"Perform additional setup for the chain (things, that cannot be done inside configuration)" looks like another (5th or 6th) config entity, we should have less of them, can it go into the simnet code with certain API points or feature flags?

I'm not suggesting doing everything in this PR, ofc.

Reply:
I agree with you Denis, this is the ideal case. In order to do this we need to do a bit more development and testing. 
This will delay a bit the move of tests into the substrate repo. We need to think about the trade-offs.
For now we can live with the current state and do further iterations to improve the configs.

Reply:
yeah, just note this somewhere, please. This PR can be merged.